### PR TITLE
Support for Timber, Aluminium and Generic sections added

### LINKED ISSUE #498: Revit_Toolkit: Add Support for Timber, Aluminium and Generic sections
#### Description:

Add `Create` and `Read` support for `TimberSection`, `AluminiumSection` and `GenericSection` added in https://github.com/BHoM/BHoM/pull/654

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #498

### Test files

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FTest%20Scripts%2FRevit%5FToolkit%2FRevit%5FToolkit%2DIssue498%2DMaterialSupport).

### Additional comments

Although there is no overlap, I would still recommend not clicking _Merge_ on this one before #520 gets merged.

### CODE REVIEW

File: Engine_Revit_UI/Convert/Structure/ToBHoM/Bar.cs [hunk 1]
```diff
@@ -159,16 +159,19 @@ public static List<Bar> ToBHoMBar(this FamilyInstance familyInstance, PullSettin
...
+                else if (materialFragment is Timber)
+                    property = BH.Engine.Structure.Create.TimberSectionFromProfile(profile, materialFragment as Timber, profileName);
+                else if (materialFragment is Aluminium)
+                    property = BH.Engine.Structure.Create.AluminiumSectionFromProfile(profile, materialFragment as Aluminium, profileName);
                 else
                 {
                     familyInstance.UnknownMaterialWarning();
-                    property = BH.Engine.Structure.Create.SteelSectionFromProfile(profile, null, profileName);
+                    property = BH.Engine.Structure.Create.GenericSectionFromProfile(profile, materialFragment, profileName);
                     property.Material = materialFragment;
```
[the remark this answers is not in the document]
Comment (pawelbaran) [reply Engine_Revit_UI/Convert/Structure/ToBHoM/Bar.cs]: Fixed

**Comment by pawelbaran** [comment]
> i.e. I create a TimberSection in BHoM,
> push to Revit, if the name is custom It does not work `1. Element type has not been found for given BHoM Object. BHoM Guid: 2f303179-5e4f-47dc-8a36-3c49c83ee6b1`, and if the name is of an existing Timber section, that will be used

Unfortunately this is how Revit_Toolkit works atm... it is only capable of name matching on push: you need to first load/create an element type in your model to be able to push elements. It should be possible to make it smarter or even fully automated, but the way in which Revit handles section types, materials, material types is complex and will need a lot of effort to cover all possible (general) cases.

**Comment by pawelbaran** [comment]
Added _do-not-merge_ label to wait for #520

**Comment by FraserGreenroyd** [comment]
I'm happy for this to be merged without review from BEnv as this does not impact us, I've taken a quick look at the code and cannot foresee any issues that might impact us at this stage, so I'm happy to sit this one out :smile:

**Comment by FraserGreenroyd** [comment]
@pawelbaran #520 doesn't exist anymore, but your do-not-merge label is still on, so will leave for you to merge when you're ready :smile:

**Pull request merged**